src/Microsoft.ML.Data/Evaluators/Metrics/ConfusionMatrix.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.ML.Runtime;
+
+namespace Microsoft.ML.Data
+{
+    public sealed class ConfusionMatrix
+    {
+        public double[] PrecisionSums { get; }
+        public double[] RecallSums { get; }
+        public ReadOnlyMemory<char>[] Labels => PredictedLabelNames.ToArray(); //sefilipi: return as string[]
+
+        public double[][] ConfusionTableCounts { get; }
+
+        internal readonly List<ReadOnlyMemory<char>> PredictedLabelNames;
+        internal readonly bool Sampled;
+        internal readonly bool Binary;
+
+        private readonly IHost _host;
+
+        internal ConfusionMatrix(IHost host, double[] precisionSums, double[] recallSums, double[][] confusionTableCounts,
+            List<ReadOnlyMemory<char>> labelNames, bool sampled, bool binary)
+        {
+            _host = host;
+
+            PrecisionSums = precisionSums;
+            RecallSums = recallSums;
+            ConfusionTableCounts = confusionTableCounts;
+            PredictedLabelNames = labelNames;
+            Sampled = sampled;
+            Binary = binary;
+        }
+
+        public override string ToString() => MetricWriter.GetConfusionTableAsString(this, false);
+
+        public double GetCountForClassPair(string predictedLabel, string actualLabel)
```

Comment:
>string predictedLabel, string actualLabel [](start = 43, length = 41)

As with all the remainder, we index slots by numeric index. They may *optionally* have a string label (or not), but that is not a requirement, and what the string label is (again, if it even exists) cannot be used as any sort of primary key.